### DISCUSSION TITLE
How to generate tailwindCSS classes under certain class name?

### DISCUSSION THREAD

**Discussion by vikingmute** [tailwindlabs/tailwindcss#7803] [Help]
Hi everyone. I am integrating a component library into existing project. I want to generate tailwindCSS classnames under certain container class.
Like the code below:
```css
.component-class .text-3xl {
}
....
```
Any help is appreciated. Thanks.

**Comment by AviAvinav** [comment]
If I understand this correctly, you want to add Tailwind CSS classes to a component from the library. If that is the case, then, you can do something like:

```
.component-class {
   @apply text-3xl;
}
```

you can still use normal css in this like this:

```
.component-class {
   @apply text-3xl;
   font-weight: bold;
}
```

**Reply by vikingmute** [reply]
@AviAvinav thanks for the reply.
But I don't want only one  particular class generated under certain class name.  I want all the tailwind classes used in the component to generate in the output css file.
For example, the component file:
```jsx
<div class="component-container">
          <img
            class="w-16 h-16 rounded-full"
          >
</div>
```
output css with className prefix:
```css
.component-container .w-16 {
}
.component-container .h-16 {
}
.component-container .rounded-full {

}
...
```
This feature is similar to prefix in tailwind configuration https://tailwindcss.com/docs/configuration#prefix, but with a new parent class name.

**Reply by AviAvinav** [reply]
I get what you mean now. As far as I know, there is no feature in Tailwind CSS that can output a CSS like that. Hope it helps.
